src/products/getProduct.js

```diff
@@ -22,20 +22,26 @@ const ProductMapper = require('./ProductMapper');
 const HttpStatusCodes = require('http-status-codes');
 
 /**
- * This action searches a single Magento product by its SKU.
+ * This action searches a single Magento product by either its SKU or slug.
  *
- * @param   {string} args.MAGENTO_SCHEMA                Magento host URL schema (http or https)
- * @param   {string} args.MAGENTO_HOST                  Magento host URL
- * @param   {string} args.GRAPHQL_PRODUCT_ATTRIBUTES    The product attributes fetched by the graphQL request
- * @param   {string} args.MAGENTO_MEDIA_PATH            Magento media base path
+ * @param   {object} args                               Object of request parameters.
+ * @param   {bool}   useSlug                            If true, this function expects args.slug to exist, otherwise it will use args.id.
  * 
- * @param   {string} args.id                            The SKU of the product.
- *
  * @return  {Promise.<Product}                          A promise which resolves to a product model representation
  */
-function getProduct(args) {
-    const validator = new InputValidator(args, ERROR_TYPE);
-    validator.checkArguments().mandatoryParameter('id');
+function getProduct(args, useSlug) {
+    let param;
+    let filterKey;
+    if (useSlug) {
```

Comment:
Instead of using `useSlug` we could move the `filterKey` as parameter. The param name can be omitted if the caller of `getProduct` already set this into a generic field so we can just use `filter: '${filterKey}:"${args.filterValue}"'` in the code below.

Reply:
@mhaack I moved the `param` variable to the arguments of the function. I didn't move the `filterValue` since this would require to do the input validation in the calling function already which in turn would lead to code duplication of that part.

Reply:
LGTM 👍